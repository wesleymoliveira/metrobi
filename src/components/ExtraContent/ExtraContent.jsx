import { Typography } from '../Typography/Typography';
import * as S from './ExtraContent.styled';


export function ExtraContent() {
	return (
		<S.Wrapper data-testid='extra-content'>
			<Typography size='xxlarge'>
				Extra Content
			</Typography>
		</S.Wrapper>
	);
}
