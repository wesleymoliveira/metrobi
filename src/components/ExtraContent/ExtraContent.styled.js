import styled, { css } from 'styled-components'

export const Wrapper = styled.section`
  ${({ theme }) => css`
    background-color: ${theme.colors.extraContentBackground};
    height: 30%;
    display: flex;
    justify-content: center;
    align-items: start;
  `}
`;
