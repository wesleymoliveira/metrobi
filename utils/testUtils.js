import React from "react";
import { ThemeProvider } from "styled-components";
import { render } from "@testing-library/react";
import renderer from "react-test-renderer";

import theme from "../src/styles/theme";

const AllTheProviders = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

const customRender = (ui, options) => {
  return render(ui, { wrapper: AllTheProviders, ...options });
};

const snapshotRenderer = (component) => {
  return renderer.create(
    <ThemeProvider theme={theme}>{component}</ThemeProvider>
  );
};

export * from "@testing-library/react";

export { customRender as render, snapshotRenderer  };
