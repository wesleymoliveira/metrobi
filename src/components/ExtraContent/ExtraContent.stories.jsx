import React from 'react';
import { ExtraContent } from './ExtraContent';

const ExtraContentArgTypes = {
  label: {
    name: 'label',
    type: { name: 'string', required: false },
    description: 'overwritten description',
    table: {
      type: { 
        summary: 'something short', 
        detail: 'something really really long' 
      },
    },
    control: {
      type: 'string'
    }
  }
};

export default {
	title: 'components/ExtraContent',
	component: ExtraContent,
	argTypes: ExtraContentArgTypes
};

export const Default = (args) => 
<div style={{ maxWidth: '130rem', margin: '0 auto' }}>
  <ExtraContent {...args} />
</div>;

