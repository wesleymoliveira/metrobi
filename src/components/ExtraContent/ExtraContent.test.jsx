import { screen } from '@testing-library/react';

import { ExtraContent } from './ExtraContent';
import { render, snapshotRenderer } from '../../../utils/testUtils';

describe('<ExtraContent />', () => {
		it('should render the component', () => {
		render(<ExtraContent />);
		
		expect(screen.getByTestId(/extra-content/i)).toBeInTheDocument();
	});

	it('should match snapshot', () => {
		const tree = snapshotRenderer(<ExtraContent />).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('should match the original background-color', () => {
		const tree = snapshotRenderer(<ExtraContent />).toJSON();
		//make sure the background-color is the original one
		expect(tree).toHaveStyleRule('background-color', '#898989')
	});

});
