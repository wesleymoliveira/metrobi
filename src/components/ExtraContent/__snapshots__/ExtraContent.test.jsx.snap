// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExtraContent /> should match snapshot 1`] = `
.c1 {
  color: #FAFAFA;
  font-size: 2.8rem;
}

.c0 {
  background-color: #898989;
  -webkit-flex: 2;
  -ms-flex: 2;
  flex: 2;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
}

<section
  className="c0"
  data-testid="extra-content"
>
  <h1
    className="c1"
    color="white"
    data-testid="typography"
    size="xxlarge"
  >
    Extra Content
  </h1>
</section>
`;
